tests/tests_pytorch/demos/transformer.py

```diff
@@ -0,0 +1,25 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+from lightning.pytorch.demos import Transformer
+
+
+def test_compile_transformer():
```

Comment:
test is unrelated to the fix, but I added it just to have a compile test.

Reply:
Perhaps you want to set `fullgraph=True` to make sure it doesn't graph break

Reply:
Thanks for the suggestion. Unfortunately we get a graph break:

```
E       torch._dynamo.exc.Unsupported: data dependent operator: aten._local_scalar_dense.default
E       
E       from user code:
E          File "/Users/adrian/repositories/lightning/src/lightning/pytorch/demos/transformer.py", line 65, in forward
E           output = self.transformer(src, target, tgt_mask=mask)
E       
E       
E       You can suppress this exception and fall back to eager by setting:
E           import torch._dynamo
E           torch._dynamo.config.suppress_errors = True

```

Since this is not the focus of the PR, I'll look at it separately.